src/log-pcap.c

```diff
@@ -1077,6 +1294,109 @@ static OutputInitResult PcapLogInitCtx(ConfNode *conf)
             }
             SCLogInfo("Using log dir %s", pl->dir);
         }
+
+        const char *compression_str = ConfNodeLookupChildValue(conf,
+                "compression");
+
+        PcapLogCompressionData *comp = &pl->compression;
+        if (compression_str == NULL || strcmp(compression_str, "none") == 0) {
+            comp->format = PCAP_LOG_COMPRESSION_FORMAT_NONE;
+            comp->buffer = NULL;
+            comp->buffer_size = 0;
+            comp->file = NULL;
+            comp->pcap_buf_wrapper = NULL;
+        } else if (strcmp(compression_str, "lz4") == 0) {
+#ifdef HAVE_LIBLZ4
+            if (pl->mode == LOGMODE_SGUIL) {
+                SCLogError(SC_ERR_INVALID_YAML_CONF_ENTRY, "Compressed pcap "
+                        "logs are not possible in sguil mode");
+                exit(EXIT_FAILURE);
+            }
+            pl->compression.format = PCAP_LOG_COMPRESSION_FORMAT_LZ4;
+
+            /* Use SCFmemopen so we can make pcap_dump write to a buffer. */
+
+            comp->pcap_buf_size = sizeof(struct pcap_file_header) +
+                    sizeof(struct pcap_pkthdr) + PCAP_SNAPLEN;
+            comp->pcap_buf = SCMalloc(comp->pcap_buf_size);
```

Comment:
This leaks on exit:
```
==28381==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 262192 byte(s) in 1 object(s) allocated from:
    #0 0x7f4d2e464850 in malloc (/lib64/libasan.so.4+0xde850)
    #1 0xf1a7b7 in PcapLogInitCtx /home/jason/oisf/code/suricata/pcap-compression/src/log-pcap.c:1321
    #2 0xff446f in RunModeInitializeOutputs /home/jason/oisf/code/suricata/pcap-compression/src/runmodes.c:774
    #3 0x11c12ed in PreRunPostPrivsDropInit /home/jason/oisf/code/suricata/pcap-compression/src/suricata.c:2223
    #4 0x411e18 in main /home/jason/oisf/code/suricata/pcap-compression/src/suricata.c:2860
    #5 0x7f4d29ef2009 in __libc_start_main (/lib64/libc.so.6+0x21009)

SUMMARY: AddressSanitizer: 262192 byte(s) leaked in 1 allocation(s).
```
It needs to be free'd in PcapLogFileDeInitCtx. Sorry I didn't catch/comment on this one before.